src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -620,7 +604,7 @@ internal static string GetCookieHeader(Uri uri, CookieContainer cookies)
             // Manually add cookies.
             if (cookies != null)
             {
-                string cookieHeader = GetCookieHeader(requestMessage.RequestUri, cookies);
+                string cookieHeader = WinHttpCookieHelper.GetCookieHeader(requestMessage.RequestUri, cookies);
```

Comment:
This code could be moved within the Cookie adapter class above to encapsulate cookie processing within a single class.